features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,582 @@
+`;
+
+const Important = styled(Typography)`
+  color: red;
+  background: black;
+  display: inline-block;
+`;
+
+const Link = styled.a`
+  color: white;
+  font-size: 14px;
+`;
+
+interface DialogProps {
+  handleClose(): any;
+  isDialogShowing: boolean;
+  selectedSponsor: string | null;
+}
+
+const PositionActionsDialog = (props: DialogProps) => {
+  const { empState } = EmpState.useContainer();
+  const { dvmState } = DvmState.useContainer();
+  const { finalFee } = dvmState;
+  const { getEtherscanUrl } = Etherscan.useContainer();
+  const { contract: emp } = EmpContract.useContainer();
+  const { latestPrice } = PriceFeed.useContainer();
+  const {
+    symbol: tokenSymbol,
+    balance: tokenBalance,
+    allowance: tokenAllowance,
+    setMaxAllowance: setMaxTokenAllowance,
+  } = Token.useContainer();
+  const {
+    symbol: collSymbol,
+    balance: collBalance,
+    allowance: collAllowance,
+    setMaxAllowance: setMaxCollateralAllowance,
+  } = Collateral.useContainer();
+  const { activeSponsors } = EmpSponsors.useContainer();
+
+  const [dialogTabIndex, setDialogTabIndex] = useState<string | null>(
+    "deposit"
+  );
+  const [collateralToDeposit, setCollateralToDeposit] = useState<string>("");
+  const [minCollPerToken, setMinCollPerToken] = useState<string>("");
+  const [maxCollPerToken, setMaxCollPerToken] = useState<string>("");
+  const [maxTokensToLiquidate, setMaxTokensToLiquidate] = useState<string>("");
+  const [deadline, setDeadline] = useState<string>("");
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const {
+    priceIdentifier: priceId,
+    collateralRequirement: collReq,
+    currentTime,
+  } = empState;
+
+  const setDialogTab = (
+    event: MouseEvent<HTMLElement>,
+    newAlignment: string | null
+  ) => {
+    setDialogTabIndex(newAlignment);
+  };
+
+  const prettyBalance = (x: number) => {
+    const x_string = x.toFixed(4);
+    return utils.commify(x_string);
+  };
+
+  const prettyAddress = (x: String | null) => {
+    if (!x) return "N/A";
+    return x.substr(0, 6) + "..." + x.substr(x.length - 6, x.length);
+  };
+  if (
+    props.selectedSponsor &&
+    activeSponsors[props.selectedSponsor] &&
+    latestPrice &&
+    collReq
+  ) {
+    const sponsorPosition = activeSponsors[props.selectedSponsor];
+
+    const pendingWithdrawTimeRemaining =
+      Number(sponsorPosition.withdrawalTimestamp) - Number(currentTime);
+
+    const pendingWithdrawTimeString =
+      pendingWithdrawTimeRemaining && pendingWithdrawTimeRemaining > 0
+        ? Math.max(0, Math.floor(pendingWithdrawTimeRemaining / 3600)) +
+          ":" +
+          Math.max(0, Math.floor((pendingWithdrawTimeRemaining % 3600) / 60)) +
+          ":" +
+          Math.max(0, (pendingWithdrawTimeRemaining % 3600) % 60)
+        : "None";
+
+    const pendingTransferTimeRemaining =
```

Comment:
For now I see no reason to include transfer logic because the rest of the dApp doesn't treat a position as if it has transfer capabilities. Let's just add to another PR?

Reply:
someone might have done the transfer themselves directly. this modal is to preform liquidations. this text is hidden In most cases if there is no pending transfer anyways.